stan/math/prim/fun/acos.hpp

```diff
@@ -30,21 +30,24 @@ struct acos_fun {
  * @param x container
  * @return Arc cosine of each variable in the container, in radians.
  */
-template <typename T, typename = require_not_eigen_vt<std::is_arithmetic, T>>
+template <typename T,
+          require_not_container_st<is_container, std::is_arithmetic, T>...>
```

Comment:
`require_ ... _st<is_container, ...>` does not make sense - scalar type of something is never a container. Did you mean to use `require_not_container_vt`? Even with this change I don't understand. Actually I think even the previous requirement was wrong - it should be `require_arithmetic_vt<T>` and `require_not_eigen_t<T>` separately.

Is this now only used for scalars? Than you can use `require_arithmetic_t<T>`. In this case also `apply_scalar_unary` can be removed.

Reply:
With ```require_container_st<is_container, std::is_arithmetic, T>```, it checks whether ```T``` is a container (either ```std::vector``` or ```Eigen```) whose base scalar type is ```arithmetic```. For example, this would capture the case where ```T``` is an ```std::vector<Eigen::MatrixXd>```.

The ```require_not_container_st``` is required because ```apply_scalar_unary``` handles all other types (scalar and containers of autodiff types)

Reply:
I went over the code of `require_not_container_st`. Now this makes a bit more sense. I was confused by name, as what it does is not consistent with other requires.